src/Components/DecisionCanvas/DecisionCanvasUpdated.jsx

```diff
@@ -0,0 +1,201 @@
+    h4 {
+      color: #000;
+      font-size: 14px;
+      font-weight: bold;
+      margin-top: 0;
+    }
+  }
+  h4 {
+    font-weight: normal;
+  }
+`;
+
+const CanvasHeader = styled.div`
+  display: flex;
+  button {
+    height: fit-content;
+    padding: 10px 16px;
+    text-transform: none;
+  }
+  h2 {
+    margin-right: auto;
+  }
+`;
+
+function expandArrayReducer(expandedArr, { type, index }) {
+  return produce(expandedArr, (draftExpanded) => {
+    switch (type) {
+      case "TOGGLE":
+        draftExpanded[index] = !expandedArr[index];
+        break;
+      case "EXPAND":
+        draftExpanded[index] = true;
+        break;
+      case "EXPAND_ALL":
+        draftExpanded.fill(true);
+        break;
+      case "COLLAPSE":
+        draftExpanded[index] = false;
+        break;
+      case "COLLAPSE_ALL":
+        draftExpanded.fill(false);
+        break;
+      default:
+    }
+  });
+}
+
+function DecisionCanvasUpdated({ update, review, categoryData }) {
+  const [expandArray, dispatch] = useReducer(
+    expandArrayReducer,
+    categoryData.map(() => false)
+  );
+
+  const categoryToReviewMap = useMemo(() => {
+    if (review == null) return {};
+    return review.questionList.reduce((reviewMap, review) => {
+      reviewMap[review.id] = review;
+      return reviewMap;
+    }, {});
+  }, [review]);
+
+  return (
+    <SectionWrapper>
+      <CanvasHeader>
+        <h2>Decision Canvas</h2>
+        <Button onClick={() => dispatch({ type: "COLLAPSE_ALL" })}>
+          Collapse all
+        </Button>
+        <Spacer />
+        <Button
+          color="primary"
+          onClick={() => dispatch({ type: "EXPAND_ALL" })}
+          variant="contained"
+        >
+          Expand all
+        </Button>
+      </CanvasHeader>
+      {categoryData
+        ? categoryData.map((section, index) => (
+            <CanvasCard
+              expanded={expandArray[index]}
+              key={section.id}
+              id={"canvas_" + section.id}
+              onHeaderClick={() => dispatch({ type: "TOGGLE", index })}
+              onLinkClick={() => dispatch({ type: "EXPAND", index })}
+              review={categoryToReviewMap["canvas_" + section.id]}
+              title={section.title}
+              update={update}
+            >
+              <CardBody>
+                <h4> {section.description} </h4>
+                <div>
+                  {section.answers.map((item, i) => (
+                    <div key={i}>
+                      <div className="questions">
+                        <h4>
+                          {i +
+                            1 +
+                            ". " +
+                            item.question
+                              .replace(/-/g, ".")
+                              .replace("One.Line", "One-Line")
+                              .replace("COVID.19", "COVID-19")
+                              .replaceAll("_dot_", ".")
+                              .replaceAll("_dash_", "-")}
```

Comment:
A few things look weird here on a first pass. Are there other cases like One.Line and Covid.19 that we want to replace? If so, we might want to investigate further and resolve the issue that is causing this in the first place. Also, why are we adding "1." to the value here (lines 126 and 127)? Seems strange to me.

Reply:
The `One.Line` and `COVID.19` cases had existed before my changes, but I can double-check with @GodGreg to see if they are still needed. The `i + 1` was just to ensure that the indexing displayed for the questions started at 1 instead of 0.

Reply:
This was only needed for the old submissions because the question text was being stored as an a key. This shouldn't be necessary with the new submissions model :)

Reply:
That's awesome! Does this mean all the replace statements can be removed or only the One.Line and COVID.19 one?

Reply:
Yep I think you can remove all of them – since the questions are a string now you should be able to just use that